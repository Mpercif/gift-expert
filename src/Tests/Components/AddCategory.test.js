import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import React from 'react';
import AddCategory from "../../Components/AddCategory"


describe('Pruebas en el componente AddCategory', () => {

    const setCategory = jest.fn();
    let wraper = shallow(<AddCategory setCategory={setCategory} /> )

    beforeEach(()=>{
        jest.clearAllMocks();
        wraper = shallow(<AddCategory setCategory={setCategory} /> )
    })

    test('Debe mostrarse correctamente ', () => {
        expect({wraper}).toMatchSnapshot()
    })

    test('El valor de input debe cambiar', () => {
        const input = wraper.find('input')
        const value = 'Hola Mundo';
        input.simulate('change', { target:{ value } })
        const p = wraper.find('p')

        expect(p.text().trim()).toBe(value)
    })
    
    test('No debe postear la informacion onSubmit ', () => {
        wraper.find('form').simulate('submit', { preventDefault(){} })

        expect(setCategory).not.toHaveBeenCalled();        
    })
    
    test('Debe de llamar el setCategory y limpiar la caja de texto ', () => {

        const value = 'Hola Mundo'
        const input = wraper.find('input')
        input.simulate('change', {target: { value }})
        
        wraper.find('form').simulate('submit', {preventDefault(){}})

        
        expect(setCategory).toHaveBeenCalled()
        expect(input.text().trim() ).toBe('')
        
    })
    


    
    
})
