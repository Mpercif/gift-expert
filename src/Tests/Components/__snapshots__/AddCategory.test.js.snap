// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe mostrarse correctamente  1`] = `
Object {
  "wraper": <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>,
}
`;
